pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -45,6 +45,7 @@ type clients struct {
 type Flavor struct {
 	*flavors.Flavor
 	Baremetal bool
+	userError error // userError contains a non-critical error that is cause by incorrect usage of the flavor field
```

Comment:
It is not clear to me why the flavor is a special case that requires storing `userError`. For example, if the OS image is not found, then validation will yield a generic "NotFound" error. This is accomplished by checking that `OSImage` in the cloud info is nil. Can we not similarly check if `ci.Flavors[flavorName].Flavor == nil` to determine if the flavor was not found?